core/src/main/java/org/infinispan/configuration/cache/AsyncConfigurationBuilder.java

```diff
@@ -62,7 +61,10 @@ public AsyncConfigurationBuilder syncMarshalling() {
     * NOTE: Currently Infinispan will not use the object instance, but instead instantiate a new
     * instance of the class. Therefore, do not expect any state to survive, and provide a no-args
     * constructor to any instance. This will be resolved in Infinispan 5.2.0
+    *
+    * @deprecated superseded by JGroups bundler functionality
```

Comment:
Actually I see we don't create a new instance any more, please remove the note paragraph completely.